streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingPage.scala

```diff
@@ -17,181 +17,446 @@
 
 package org.apache.spark.streaming.ui
 
-import java.util.Calendar
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.concurrent.TimeUnit
 import javax.servlet.http.HttpServletRequest
 
-import scala.xml.Node
+import scala.collection.mutable.ArrayBuffer
+import scala.xml.{Node, Unparsed}
 
 import org.apache.spark.Logging
 import org.apache.spark.ui._
 import org.apache.spark.ui.UIUtils._
-import org.apache.spark.util.Distribution
+
+/**
+ * @param timelineDivId the timeline `id` used in the html `div` tag
```

Comment:
Just add a line explaining what this data structure represents. Also for other **UIData files.